frontend/src/newsfeed/list/NewsfeedList.tsx

```diff
@@ -152,6 +193,28 @@ const NewsfeedList: React.FC<NewsfeedListProps> = ({
             });
     };
 
+    const setFiltersWrapper = (proposedFilters: string[]) => {
+        const addedFilters = proposedFilters.filter(
+            (filter) => !filters.includes(filter),
+        );
+
+        let finalFilters = [];
+        if (addedFilters.includes(AllCategoriesFilterName)) {
+            finalFilters = [AllCategoriesFilterName];
+        } else {
+            finalFilters = proposedFilters.filter(
+                (filter) => filter != AllCategoriesFilterName,
+            );
+        }
+
+        setFilters(finalFilters);
+    };
+
+    const allEntries = data?.entries ?? [];
+    const shownEntries = allEntries.filter((entry) =>
```

Comment:
Can we check that showAdditionalFilters is true before performing this operation? If it's false, then shownEntries will always equal allEntries and there isn't any reason to loop through all the entries.